homeassistant/components/brother/sensor.py

```diff
@@ -66,10 +81,30 @@ def state(self):
     def device_state_attributes(self):
         """Return the state attributes."""
         if self.kind == ATTR_DRUM_REMAINING_LIFE:
-            self._attrs["remaining_pages"] = self.printer.data.get(
+            self._attrs[ATTR_REMAINING_PAGES] = self.printer.data.get(
                 ATTR_DRUM_REMAINING_PAGES
             )
-            self._attrs["counter"] = self.printer.data.get(ATTR_DRUM_COUNTER)
+            self._attrs[ATTR_COUNTER] = self.printer.data.get(ATTR_DRUM_COUNTER)
+        elif self.kind == ATTR_BLACK_DRUM_REMAINING_LIFE:
+            self._attrs[ATTR_REMAINING_PAGES] = self.printer.data.get(
+                ATTR_BLACK_DRUM_REMAINING_PAGES
+            )
+            self._attrs[ATTR_COUNTER] = self.printer.data.get(ATTR_BLACK_DRUM_COUNTER)
+        elif self.kind == ATTR_CYAN_DRUM_REMAINING_LIFE:
+            self._attrs[ATTR_REMAINING_PAGES] = self.printer.data.get(
+                ATTR_CYAN_DRUM_REMAINING_PAGES
+            )
+            self._attrs[ATTR_COUNTER] = self.printer.data.get(ATTR_CYAN_DRUM_COUNTER)
+        elif self.kind == ATTR_MAGENTA_DRUM_REMAINING_LIFE:
+            self._attrs[ATTR_REMAINING_PAGES] = self.printer.data.get(
+                ATTR_MAGENTA_DRUM_REMAINING_PAGES
+            )
+            self._attrs[ATTR_COUNTER] = self.printer.data.get(ATTR_MAGENTA_DRUM_COUNTER)
+        elif self.kind == ATTR_YELLOW_DRUM_REMAINING_LIFE:
+            self._attrs[ATTR_REMAINING_PAGES] = self.printer.data.get(
+                ATTR_YELLOW_DRUM_REMAINING_PAGES
+            )
+            self._attrs[ATTR_COUNTER] = self.printer.data.get(ATTR_YELLOW_DRUM_COUNTER)
```

Comment:
Using the local variables for updating `self._attrs`. Therefore, this block of code has to only be written once and not multiple times for each color making the code easier to read IMHO.

What do you think? 😊

```suggestion
            self._attrs[ATTR_COUNTER] = self.printer.data.get(ATTR_YELLOW_DRUM_COUNTER)
        if remaining_pages and drum_counter:
            self._attrs[ATTR_REMAINING_PAGES] = self.printer.data.get(remaining_pages)
            self._attrs[ATTR_COUNTER] = self.printer.data.get(drum_counter)
```

Reply:
@springstan Great idea! Change done.

Reply:
@bieniu thank you ✌